### ISSUE TITLE
Clarify what is being tested with `L003's` `lint_templated_tokens` option

### ISSUE THREAD

**Issue by jgillies** [sqlfluff/sqlfluff#1533]
The following templated SQL produces a `L003` failure when `lint_templated_tokens = False`:

```sql
SELECT

    NAME                                                      AS quote_job_name,
    {{ convert_sf_timezone("CREATED_DATE") }}                 AS created_ts,
    CREATE_BY_ID                                              AS create_by_user_id,
...
```

The reason for the failure is that the rendered SQL is as follows:

```sql
SELECT

    NAME                                                      AS quote_job_name,
    
TIMESTAMP(DATETIME_SUB(CREATED_DATE, INTERVAL 4 HOUR), 'America/Los_Angeles')
                 AS created_ts,
    CREATE_BY_ID                                              AS create_by_user_id,
...
```

The rendered SQL is being properly linted, but this might not be the expected behavior from the perspective of the person writing the templated SQL, where they are more interested in their written SQL being linted than the rendered SQL. The docs for `lint_templated_tokens` don't make it entirely clear whether the example templated SQL should pass or fail.

As a side note, trimming whitespace in the `convert_sf_timezone()` macro actually solved the issue, as the rendered SQL was properly indented.

**Comment by barrywhart** [comment]
If it was a useful feature and we know what it's supposed to do, we. could potentially restore it, but I did remove it because it was causing problems with an earlier issue, and with no tests or docs for it, it seemed simplest to remove it at that time.

**Comment by tunetheweb** [comment]
I think the better solution it to follow L016's example and make L003 only check indentation of the raw, pre-templated code - that's the code people will be looking at and working on, not the post templated code.
